### ISSUE TITLE
Text execution hangs on the click action

### ISSUE THREAD

**Issue by aleks-pro** [DevExpress/testcafe-hammerhead#2514]
### Steps to Reproduce:
1. Clone the https://github.com/javenkn/testcafe-poc repository
2. Run the `npm install` command
3. Run the `test` script from `package.json` file

The test fails on the [`navigate to checkout`](https://github.com/javenkn/testcafe-poc/blob/main/tests/page.test.js#L36) step even if an element with the `'button[data-test="qa-cart-checkout"]'` selector appears. It looks like TestCafe waites for AJAX requests' completion before a click.

### Your Environment details:
* node.js version: v12.18.0
* browser name and version: Chrome 87.0.4280.88
* platform and version: macOS 10.15.6

**Comment by shyedhu** [comment]
Thank you @aleks-pro for creating this issue

**Comment by shyedhu** [comment]
Hi @aleks-pro , Since we are blocked few e2e automation test runs due to that 222 response code, so can you please help to provide us an update for 2514. Thanks in advance !

**Comment by arubtsov** [comment]
@shyedhu, we still need time to research this issue. Once we get any results, we will post them here.

**Comment by shyedhu** [comment]
Hi @arubtsov Thank you for the update !

**Comment by Farfurix** [comment]
@shyedhu 

Hello,

This issue has been resolved in `testcafe@1.10.2-alpha.1`. So, I'll close this issue.

**Comment by shyedhu** [comment]
@Farfurix Thanks for your help ! 222 response code issue has been resolved , but it throws 403 error code which blocked my other test case execution
=============
body:
        '<HTML><HEAD>\n<TITLE>Access Denied</TITLE>\n</HEAD><BODY>\n<H1>Access Denied</H1>\n \nYou don\'t have permission to access "http....

================

Can you please help to provide the solution to avoid 403 status code and how can we get right status code like 202 during my test run . i don't see kind of issue in other automation framework like nightwatchjs 

I tried to set correct origin value on Access-Control-Allow-Origin header in CustRequestHook class and then add this into  .requestHooks(logger,custRequestHook) in page.test.js  , but it still throwing 403 error code ,Can you please help me on these?

```
import { RequestHook } from 'testcafe';

export default class CustRequestHook extends RequestHook {
  constructor (requestFilterRules, responseEventConfigureOpts) {
    super(requestFilterRules, responseEventConfigureOpts);
    // ...
}

  onResponse(event) {

event.headers['Access-Control-Allow-Origin']='xxxxxx';
  }
}
```

**Comment by Farfurix** [comment]
@shyedhu 

Hello,

> but it throws 403 error code which blocked my other test case execution

Could you please create a separate issue with a reproducible example? We need to examine your case on our side.

**Comment by shyedhu** [comment]
HI @Farfurix  Here's new issue for your reference. https://github.com/DevExpress/testcafe-hammerhead/issues/2565 
Please help us to unblock our e2e test execution . Thanks in advance
